docs/lorax-composer.rst

```diff
@@ -4,61 +4,198 @@ lorax-composer
+* SELinux must be set to Permissive or disabled to allow ``livemedia-creator`` to run Anaconda.
+* Only allow authorized users access to the ``weldr`` group and socket.
+
+Since Anaconda kickstarts are used there is the possibility that a user could
+inject commands into a blueprint that would result in the kickstart executing
+arbitrary code on the host.  Only authorized users should be allowed to build
+images using ``lorax-composer``.
+
+How it Works
+------------
+
+The server runs as root, and as ``weldr``. Communication with it is via a unix
+domain socket (``/run/weldr/api.socket`` by default). The directory and socket
+are owned by ``root:weldr`` so that any user in the ``weldr`` group can use the API
+to control ``lorax-composer``.
+
+At startup the server will check for the correct permissions and
+ownership of a pre-existing directory, or it will create a new one if it
+doesn't exist.  The socket path and group owner's name can be changed from the
+cmdline by passing it the ``--socket`` and ``--group`` arguments.
+
+It will then drop root privileges for the API thread and run as the ``weldr``
+user. The queue and compose thread still runs as root because it needs to be
+able to mount/umount files and run Anaconda.
 
 Composing Images
 ----------------
 
-As of version 19.7.7 lorax-composer can create ``tar`` output images. You can use curl to start
-a compose like this::
+The `welder-web <https://github.com/weldr/welder-web/>`_ GUI project can be used to construct
+blueprints and create composes using a web browser.
+
+Or use the command line with `composer-cli <composer-cli.html>`_.
+
+Blueprints
+----------
+
+Blueprints are simple text files in `TOML <https://github.com/toml-lang/toml>`_ format that describe
+which packages, and what versions, to install into the image. They can also define a limited set
+of customizations to make to the final image.
+
+Example blueprints can be found in the ``lorax-composer`` `test suite
+<https://github.com/weldr/lorax/tree/master/tests/pylorax/blueprints/>`_, with a simple one
+looking like this::
+
+    name = "base"
+    description = "A base system with bash"
+    version = "0.0.1"
+
+    [[packages]]
+    name = "bash"
+    version = "4.4.*"
+
+The ``name`` field is the name of the blueprint. It can contain spaces, but they will be converted to ``-``
+when it is written to disk. It should be short and descriptive.
+
+``description`` can be a longer description of the blueprint, it is only used for display purposes.
+
+``version`` is a `semver compatible <https://semver.org/>`_ version number. If
+a new blueprint is uploaded with the same ``version`` the server will
+automatically bump the PATCH level of the ``version``. If the ``version``
+doesn't match it will be used as is. eg. Uploading a blueprint with ``version`` set to ``0.0.1`` when the existing blueprint ``version`` is ``0.0.1`` will result in the new blueprint being stored as ``version 0.0.1``.
+
```

Comment:
I think you meant this to say something like "eg. Uploading a blueprint with version set to 0.1.0 when the existing blueprint version is 0.0.1 will result in the new blueprint being stored as version 0.1.0."  That is, the first and last version numbers shouldn't be the same as the middle.

Reply:
oh, ha, exactly :)